### PR TITLE
[TIMOB-24805] iOS: Don't require large window-titles value to be set

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24805

Unit-tests included.

### CODE REVIEW

**Review by ssjsamir — APPROVED** [review]
FR Passed: No error log was shown. 

Test Steps: 
+ Created a new titanium project 
+ Added the following code in to the project
```
var win = Ti.UI.createWindow( {
backgroundColor : 'white'
});
 
var btn = Ti.UI.createButton({
title: 'Trigger'
});
 
btn.addEventListener('click', openWindow);
win.add(btn);
 
var nav = Ti.UI.iOS.createNavigationWindow({
window: win
});
 
nav.open();
 
function openWindow() {
var win = Ti.UI.createWindow( {
backgroundColor : 'white'
} );
 
nav.openWindow(win);
}
```
+ Ran the application from studio
+ No error was shown in Studio

**Test Environment**
Appcelerator Command-Line Interface, Version 6.3.0-master.15
iPhone 7 emulator (11.0)
iphone 5S plus 11.0.3
Operating System Name: Mac OS High Sierra 
Operating System Version: 10.13
Node.js Version: 6.10.1
Xcode: 9.0
Appcelerator Studio: 4.10.0.201709271713

**Comment by mukherjee2** [comment]
@hansemannn can you please resolve conflicts and merge?

File: tests/Resources/ti.ui.window.test.js [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+
+	it('#applyProperties(Object)', function () {
+		win = Ti.UI.createWindow();
+		win.open();
+		should(win.custom).not.exist;
+		win.applyProperties({ custom: 1234 });
+		should(win.custom).be.eql(1234);
+	});
+	
+	it.ios('largeTitleEnabled', function () {
```
Comment (sgtcoolguy) [inline tests/Resources/ti.ui.window.test.js]: FYI To make things easier for merging in new tests, you can now just add a `tests/Resources/ti.ui.window.addontest.js` with just the new tests and it'll get auto-picked up (and I don't have to wade through what changed between this overriding file and the original.).

Comment (hansemannn) [reply tests/Resources/ti.ui.window.test.js]: This is awesome news Chris, thanks! 🙂

**Pull request merged**